test/e2e/autotls/http01/http01_auto_tls_test.go

```diff
@@ -0,0 +1,285 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package http01
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/test/ingress"
+	routenames "knative.dev/serving/pkg/reconciler/route/resources/names"
+	"knative.dev/serving/test"
+	testingress "knative.dev/serving/test/conformance/ingress"
+	"knative.dev/serving/test/e2e"
+	autotls "knative.dev/serving/test/e2e/autotls"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const dnsRecordDeadlineSec = 600 // 10 min
+
+type dnsRecord struct {
+	ip     string
+	domain string
+}
+
+type config struct {
+	AutoTLSDomain                 string `envconfig:"auto_tls_domain" required:"false"`
+	DNSZone                       string `envconfig:"dns_zone" required:"false"`
+	CloudDNSServiceAccountKeyFile string `envconfig:"cloud_dns_service_account_key_file" required:"false"`
+	CloudDNSProject               string `envconfig:"cloud_dns_project" required:"false"`
+	SetUpDNS                      string `envconfig:"set_up_dns" required:"false"`
+}
```

Comment:
DNS shouldn't be configured as part of the text, it should be configured as a part of setup.  I should not need Google credentials to run the e2e tests.

Reply:
I removed this into the test configuration.